File: property_unit_sort.exs
#!/usr/bin/env elixir

#filename = "units.txt"
filename = List.first(System.argv) || raise "Enter a filename"
{:ok, units_bitstring} = File.read filename
units = String.split(units_bitstring, "\n", trim: true)

apartment_regex = ~r/^#?(?<num>\d+)(?<sub_id>\w*).*/

apartment_unit_sorter = fn unit_string ->
  match = Regex.named_captures(apartment_regex, unit_string)
  num = elem(Integer.parse(Map.get(match, "num")), 0)
  sub_id = Map.get(match, "sub_id")

  [num, sub_id]
end

sorted_apartments = Enum.sort_by(units, apartment_unit_sorter)
Enum.each sorted_apartments, &IO.puts/1

File: property_unit_sort.rb
#!/usr/bin/env ruby

apartment_pattern = /^#?(\d+)(\w*).*/

filename = ARGV[0] or raise "You must specify a filename"
apartments = File.read(filename).split("\n")

sorted_apartments = apartments.sort_by do |apartment|
  match = apartment_pattern.match(apartment)
  [match[1].to_i, match[2]]
end

sorted_apartments.each { |apartment| puts apartment }
